src/docs/development/packages-and-plugins/plugin-api-migration.md

```diff
@@ -32,70 +32,51 @@ a plugin that uses the new APIs, see the
-   <br><br>
    In addition, you should document all non-overridden public members within the
    plugin. In an add-to-app scenario, these classes will be accessible to a
    developer and require documentation.
 
-2. (Optional) If your plugin needs an `Activity` reference,
-   also implement `ActivityAware`.
+1. (Optional) If your plugin needs an `Activity` reference, also implement
+   the [`ActivityAware`][] interface.
 
-3. (Optional) If your plugin is expected to be held in a
-   background Service at any point in time,
-   implement `ServiceAware`.
-
-4. Update the example app's `MainActivity.java` to use the
-   v2 embedding FlutterActivity. You may have to make a
-   public constructor for you plugin class if one didn't
-   exist already. For example:
-
-```java
-package io.flutter.plugins.firebasecoreexample;
+1. (Optional) If your plugin is expected to be held in a background Service at
+   any point in time, implement the [`ServiceAware`][] interface.
 
-import io.flutter.embedding.android.FlutterActivity;
-import io.flutter.embedding.engine.FlutterEngine;
-import io.flutter.plugins.firebase.core.FirebaseCorePlugin;
+1. Update the example app's `MainActivity.java` to use the
+   v2 embedding FlutterActivity. See [flutter.dev/go/android-project-migration][]
+   for details. You may have to make a public constructor for you plugin class
+   if one didn't exist already. For example:
 
-public class MainActivity extends FlutterActivity {
-  // TODO(<github-username>): Remove this once v2 of
-  // GeneratedPluginRegistrant rolls to stable.
-  // https://github.com/flutter/flutter/issues/42694
-  @Override
-  public void configureFlutterEngine(FlutterEngine flutterEngine) {
-    flutterEngine.getPlugins().add(new FirebaseCorePlugin());
-  }
-}
-```
+   ```java
+    package io.flutter.plugins.firebasecoreexample;
 
-5. (Optional) Use ShimPluginRegistry to add plugins that don’t yet
-   support the v2 embedding. For example:
+    import io.flutter.embedding.android.FlutterActivity;
+    import io.flutter.embedding.engine.FlutterEngine;
+    import io.flutter.plugins.firebase.core.FirebaseCorePlugin;
 
-```java
-ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
-PathProviderPlugin.registerWith(
-        shimPluginRegistry.registrarFor("io.flutter.plugins.pathprovider.PathProviderPlugin"));
-VideoPlayerPlugin.registerWith(
-        shimPluginRegistry.registrarFor("io.flutter.plugins.videoplayer.VideoPlayerPlugin"));
-```
+    public class MainActivity extends FlutterActivity {
+      // You can keep this empty class or remove it. Plugins on the new embedding
+      // now automatically registers plugins.
+    }
+    ```
 
-6. Create an `EmbeddingV1Activity.java` file that uses the v1
-   embedding in the same folder as `MainActivity`. For example:
+1. (Optional) Create an `EmbeddingV1Activity.java` file that uses the v1
+   embedding in the same folder as `MainActivity` in the example project to keep
```

Comment:
Awkward.  "Create an `EmbeddingV1Activity.java` file that uses the v1 embedding for the example project in the same folder as `MainActivity` to keep..."

Reply:
ok